llm-finetuning/huggingface/hf_deployment_base_config.py

```diff
@@ -0,0 +1,25 @@
+from pydantic import BaseModel
+from typing import Optional, Dict
+from zenml.utils.secret_utils import SecretField
+
+
+class HuggingFaceBaseConfig(BaseModel):
+    """Huggingface Inference Endpoint configuration."""
+
+    endpoint_name: Optional[str] = None
+    repository: Optional[str] = None
+    framework: Optional[str] = None
+    accelerator: Optional[str] = None
+    instance_size: Optional[str] = None
+    instance_type: Optional[str] = None
+    region: Optional[str] = None
+    vendor: Optional[str] = None
```

Comment:
This is nitpick, but as these are now optional, there is no way that the user really knows what to pass in at the step level. Do you think it makes sense to rework this "BaseConfig" idea and just make seperate configs for the deployer and the service, so we can mark things as optional or not?